### PR TITLE
 Backport HostedForm device data collection to v2.x

### PR DESCRIPTION
**Pull request by gsmendoza**
## Summary

Closes https://github.com/solidusio/solidus_braintree/issues/132.

## Checklist

Check out our [PR guidelines](https://github.com/solidusio/.github/blob/master/CONTRIBUTING.md#pull-request-guidelines) for more details.

The following are mandatory for all PRs:

- [x] I have written a thorough PR description.
- [x] I have kept my commits small and atomic.
- [x] [I have used clear, explanatory commit messages](https://github.com/solidusio/.github/blob/main/CONTRIBUTING.md#writing-good-commit-messages).

The following are not always needed:

- 📖 I have updated the README to account for my changes.
- 📑 I have documented new code [with YARD](https://www.rubydoc.info/gems/yard/file/docs/Tags.md).
- 🛣️ I have opened a PR to update the [guides](https://github.com/solidusio/edgeguides).
- ✅ I have added automated tests to cover my changes.
- 📸 I have attached screenshots to demo visual changes.

### CODE REVIEW

File: app/models/solidus_braintree/transaction.rb [hunk 1]
```diff
@@ -17,7 +17,7 @@ class Transaction
       unless payment_method.is_a? SolidusBraintree::Gateway
         errors.add(:payment_method, 'Must be braintree')
       end
-      if address && !address.valid?
+      if address&.invalid?
```
Comment (gsmendoza) [inline app/models/solidus_braintree/transaction.rb:20]: I added these changes to fix the RuboCop offenses in CI.

File: app/models/solidus_braintree/transaction_import.rb [hunk 2]
```diff
@@ -9,9 +9,9 @@ class InvalidImportError < StandardError; end
     include ActiveModel::Model
 
     validate do
-      errors.add("Address", "is invalid") if address && !address.valid?
+      errors.add("Address", "is invalid") if address&.invalid?
```
Comment (gsmendoza) [inline app/models/solidus_braintree/transaction_import.rb:12]: I added these changes to fix the RuboCop offenses in CI.

**Review by kennyadsl — APPROVED** [review]
Approving, but I have a doubt: is this supposed to be part of 2.0? I thought we wanted to leave the 2.0 version as identical as possible to SolidusPaypalBraintree to facilitate the upgrade. If we add other features, maybe it will be more complex to upgrade. 

It's not the end of the world but we can release this change in a 2.1 version. WDYT?

**Comment by gsmendoza** [comment]
@kennyadsl Makes sense. Plus, I think it would be ideal if we sort out the remaining concerns with device data (compatibility with other payment methods, reuse with saved payment sources) before we backport it to 2.0x. I'm closing this for now. We can reopen it once we're ready.

**Pull request closed without merging**